buildpack.md

```diff
@@ -91,6 +80,54 @@ Buildpack API versions:
  - `<major>` and `<minor>` MUST only contain numbers (unsigned 64 bit integer)
  - When `<major>` is greater than `0` increments to `<minor>` SHALL exclusively indicate additive changes
 
+## Terminology
+
+### CNB Terminology
+
+A **buildpack** is a directory containing a `buildpack.toml`. A buildpack analyzes application source code and determines the best way to build it.
+
+A **buildpack group**, or **group**, is a list of one or more buildpacks that are designed to work together - for example, a buildpack that provides `node` and a buildpack that provides `npm`.
+
+An **order** is a list of one or more groups to be tested against application source code, so that the appropriate group for a build can be determined. 
+
+An **executable buildpack** is a buildpack containing `/bin/detect` and `/bin/build` executables.
+
+An **order buildpack** is a buildpack containing an order definition in `buildpack.toml`. Order buildpacks do not contain `/bin/detect` or `/bin/build` executables.
+
+**Resolving an order** is the process by which an order (which may contain order buildpacks) is evaluated together with application source code to produce a group of executable buildpacks that can be used to build the application. This process is known as **detection**. During detection, the `/bin/detect` executable for each executable buildpack is invoked, and the group that is returned is the first group where `/bin/detect` returns true for all non-optional executable buildpacks.
+
+A **lifecycle** is software that orchestrates a build. It executes in a series of phases that each have a distinct responsibility:
+  1. **Detection,** where an optimal selection of compatible buildpacks is chosen by resolving the provided order.
+  2. **Analysis,** where metadata about OCI layers generated during a previous build are made available to executable buildpacks in the selected group.
+  3. **Build,** where the `bin/build` for each executable buildpack in the selected group is invoked, in order, to build the application.
+  4. **Export,** where filesystem changes from the build phase are packaged into layers in an OCI image.
+
+A **launcher** is an executable that is the `ENTRYPOINT` of the exported OCI image. It is used to start processes at runtime. Having a launcher enables multiple process types to be defined on an image, with process-specific environment variables and other functionality.
+
+**Launch** describes the process of running an application by creating a container from the exported OCI image.
+
+A **platform** is a system or software that orchestrates the lifecycle by invoking each lifecycle phase in order.
+
+A **process type** is a definition, provided by executable buildpacks during the build phase, of a process to launch at runtime.
```

Comment:
Feedback from 4/21 Working Group: we should change `process type` to `process definition` and only use `process type` when we are talking about the specific key `type`.